src/AppBundle/Entity/Report/Report.php

```diff
@@ -1159,4 +1160,102 @@ public function setChecklist($checklist)
     {
         $this->checklist = $checklist;
     }
+
+        /**
+         * Previous report data. Just return id and type for second api call to allo new JMS groups
+         *
+         * @JMS\VirtualProperty
+         * @JMS\SerializedName("previous_report_data")
+         * @JMS\Groups({"previous-report-data"})
+         * @JMS\Type("array")
+         *
+         * @return array
+         */
+        public function getPreviousReportData()
+        {
+            $previousReport = $this->getPreviousReport();
+
+            if (empty($previousReport)) {
+                return false;
+            }
+
+            return [
+                'report-summary' => $previousReport->getReportSummary(),
+                'financial-summary' => $previousReport->getFinancialSummary()
+            ];
+        }
+
+    /**
+     * Method to identify and return previous report.
+     *
+     * @return Ndr|Report|bool|mixed
+     */
+    private function getPreviousReport() {
+        $clientReports = $this->getClient()->getReports();
```

Comment:
alternative could be a query with client=same client, report!=currentTeport and taking the max(id) ,but not sure it's simpler

Reply:
or maybe, just iterate client reports excluding the current, index by reportId, then krsort, then take the first (array_shift)